packages/twenty-front/src/modules/settings/roles/components/SettingsRolesList.tsx

```diff
@@ -33,26 +43,68 @@ const StyledNoRoles = styled(TableCell)`
 
 export const SettingsRolesList = () => {
   const navigateSettings = useNavigateSettings();
+  const activeTabId = useRecoilComponentValue(
+    activeTabIdComponentState,
+    ROLES_LIST_TABS.COMPONENT_INSTANCE_ID,
+  );
 
   const settingsAllRoles = useRecoilValue(settingsAllRolesSelector);
 
   const sortedSettingsAllRoles = [...settingsAllRoles].sort((a, b) =>
     sortByAscString(a.label, b.label),
   );
 
+  const filteredRoles = sortedSettingsAllRoles.filter((role) => {
+    switch (activeTabId) {
+      case ROLES_LIST_TABS.TABS_IDS.USER_ROLES:
+        return role.canBeAssignedToUsers;
+      case ROLES_LIST_TABS.TABS_IDS.AGENT_ROLES:
+        return role.canBeAssignedToAgents;
+      case ROLES_LIST_TABS.TABS_IDS.API_KEY_ROLES:
+        return role.canBeAssignedToApiKeys;
+      default:
+        return role.canBeAssignedToUsers;
+    }
+  });
+
+  const tabs = [
+    {
+      id: ROLES_LIST_TABS.TABS_IDS.USER_ROLES,
+      title: t`User Roles`,
+      Icon: IconUser,
+    },
+    {
```

Comment:
How does this behave if I don't have the AI flag enabled? let's make sure we don't break the app / don't show the AI-related stuff now